src/plugins/dashboard/public/application/utils/use/use_dashboard_container.tsx

```diff
@@ -111,6 +114,9 @@ const createDashboardEmbeddable = (
     http,
     dashboardConfig,
     embeddableCapabilities,
+    notifications,
```

Comment:
There are some conplex functions here, for example this `createDashboardEmbeddable`.  We could decouple unrelated logic, like create helper functions for unrelated pieces of logic that can be isolated. For example, getDashboardInput and getEmptyScreenProps and also inputSubscription and outSubscription handling. The logic then could much simplized

```
const createDashboardEmbeddable = async (
 dashboardFactory,
  some params
) => {
  // Replaced the Promise chain with async/await
  const container = await dashboardFactory?.create(getDashboardInput());

  if (container && !isErrorEmbeddable(container)) {
    container.renderEmpty()
    container.getChangesFromAppStateForContainerState()
    const { inputSubscription, outputSubscription } = createSubscriptions(container, appState, dashboardServices, filterManager, queryStringManager);
    // ... do something with the subscriptions, or store them elsewhere
  }

  return container;
};
```
But I understand this is a processing PR and we will have a final polish one. Document these functions with import logic would also be helpful.

Reply:
yea `createDashboardEmbeddable` does seem like an overloaded function, can revisit later to see if possible refactor can be done